src/basic/os-util.c

```diff
@@ -460,7 +460,7 @@ int os_release_support_ended(const char *support_end, bool quiet, usec_t *ret_eo
                                       "Failed to parse SUPPORT_END= in os-release file, ignoring: %m");
 
         time_t eol = timegm(&tm);
-        if (eol == (time_t) -1)
```

Comment:
hmm, are you sure timegm() can return valid negative values? the man page doesn't mention that. and this certainly would create an ambiguity regarding one second before the epoch?

Reply:
looking at glibc sources, there appear to be no checks regarding this. we might need more than this fix though

Reply:
Yes:

```c
#include <inttypes.h>
#include <stdio.h>
#include <string.h>
#include <sys/time.h>
#include <time.h>

int main(void) {
	const char *support_end = "1960-01-01";
	struct tm tm = {};
        const char *k = strptime(support_end, "%Y-%m-%d", &tm);
        if (!k || *k)
                return -1;
        time_t eol = timegm(&tm);
	printf("Value is: %" PRId64 "\n", eol);
	return 0;
}
```

yields:

```
Value is: -315619200
```

on both:

```
ld.so (Gentoo 2.39-r6 (patchset 6)) stable release version 2.39.
```

and

```
musl libc (powerpc64) Version 1.2.3
```

I agree it does result in ambiguity with one-second-before 1970-01-01T00:00:00Z.